packages/flutter/lib/src/material/checkbox.dart

```diff
@@ -488,6 +488,10 @@ class _CheckboxState extends State<Checkbox> with TickerProviderStateMixin, Togg
               value: value,
               tristate: tristate,
               onChanged: onChanged,
+              mouseCursor: WidgetStateProperty.resolveWith((Set<WidgetState> states) {
+                return WidgetStateProperty.resolveAs<MouseCursor?>(widget.mouseCursor, states)
+                  ?? SystemMouseCursors.basic;
+              }),
```

Comment:
I'm still confused about this, I'm leaving a comment on https://github.com/flutter/flutter/pull/151788#discussion_r1683983759.